### ISSUE TITLE
Finite difference classes have no example

### ISSUE THREAD

**Issue by mbaudin47** [openturns/openturns#1757]
There are several classes related to finite difference computational methods which do not have examples, neither in the API doc nor in the Examples doc:
* http://openturns.github.io/openturns/master/user_manual/_generated/openturns.FiniteDifferenceGradient.html
* http://openturns.github.io/openturns/master/user_manual/_generated/openturns.FiniteDifferenceStep.html
* http://openturns.github.io/openturns/master/user_manual/_generated/openturns.ConstantStep.html
* http://openturns.github.io/openturns/master/user_manual/_generated/openturns.BlendedStep.html

These examples proved to be useful for some users, e.g.: 

https://openturns.discourse.group/t/sorm-vs-pythonfunction/88

Moreover, there is no example to set the gradient and use it for Taylor expansion. An example is provided here:

https://stackoverflow.com/questions/59096111/how-to-estimate-the-taylor-decomposition-variance-from-a-python-function-with-us

that may be included into the doc.

@JPelamatti Would you be interested in creating one or several examples on this topic?

**Comment by jschueller** [comment]
We have 22 issues about missing examples
https://github.com/openturns/openturns/issues?q=is%3Aissue+is%3Aopen+%22no+example%22
To reduce noise I suggest to gather these in a single issue (use the checkboxes)

**Comment by mbaudin47** [comment]
I did not think of this way. I rather do not, however, use this method because each issue page allows to add examples and discuss, before the actual example is created. If we put all the content into a single page, we will not be able to see which message is for which issue. Finally, this might be a never-closed issue, because newer missing examples will constantly add to the list. Whatever my point of view, if that method is advocated among the developers, I will follow the rule. What do you think @josephmure ?

**Comment by josephmure** [comment]
Indeed, it is useful to have issues serve as separate reminders of missing documentation. I find it practical in the context of doc-related training sessions at EDF to have 1 issue per missing example, because I can then assign exercises such as "Fix issue #xxxx". Moreover, when detailed suggestions are provided by @mbaudin47 about how an issue can be fixed (like code snippets), then that issue can be easily assigned to unexperienced doc developers. It would be tough to provide such suggestions in a large thread.

However, I agree with @jschueller that we could use fewer separate doc-related pull requests. The GitHub online interface makes it possible to write more than one commit before submitting a pull request.

![image](https://user-images.githubusercontent.com/31989332/107409241-3e10cc80-6b0c-11eb-9bf3-093a6293a9d9.png)

**Comment by mbaudin47** [comment]
Ok, this sounds good. Say I have fixed 5 bugs in a branch on my repo: how may other developers know that each bug is currently being developed? May I assign the bug to myself or write a message in the issue thread?

**Comment by josephmure** [comment]
I think if you write "Fix #xxxx" in a commit message, GitHub detects it and signals it in the issue thread even if the commit exists only on your fork. Moreover, nothing prevents you from submitting a PR even if you have only fixed 1 bug yet. You can always make it a draft PR until you have enough doc bug fixes (say, 5) to make it worthwhile for reviewers to compile your branch and review all of them at once.

**Comment by mbaudin47** [comment]
Good for me. I will try this.

**Comment by mbaudin47** [comment]
@jschueller : could you please assign this issue to @JPelamatti , who is currently working on this topic?

**Comment by jschueller** [comment]
ok

**Comment by mbaudin47** [comment]
Thank you !

**Comment by mbaudin47** [comment]
@JPelamatti : The bug was partly fixed in #1811. 
I have just a slightly comment. Would you please insert text comments in the current API example of `BlendedStep` below?

![image](https://user-images.githubusercontent.com/31351465/115733617-54979780-a389-11eb-89be-6804f3a84093.png)
